src/api/Feed.js

```diff
@@ -518,13 +519,17 @@ class Feed extends Base {
 
         const versionsPromise = shouldShowVersions ? this.fetchVersions() : Promise.resolve();
         const currentVersionPromise = shouldShowVersions ? this.fetchCurrentVersion() : Promise.resolve();
+
+        const filteredActivityTypes = [];
+        if (shouldShowAnnotations && permissions[PERMISSION_CAN_VIEW_ANNOTATIONS]) {
+            filteredActivityTypes.push(FILE_ACTIVITY_TYPE_ANNOTATION);
+        }
+        if (shouldShowAppActivity) filteredActivityTypes.push(FILE_ACTIVITY_TYPE_APP_ACTIVITY);
+        filteredActivityTypes.push(FILE_ACTIVITY_TYPE_COMMENT);
+        if (shouldShowTasks) filteredActivityTypes.push(FILE_ACTIVITY_TYPE_TASK);
```

Comment:
not sure if this is more readable but:
```
const filteredActivityTypes = [
    FILE_ACTIVITY_TYPE_COMMENT,
    ...(shouldShowAnnotations && permissions[PERMISSION_CAN_VIEW_ANNOTATIONS] ? [FILE_ACTIVITY_TYPE_ANNOTATION] : []),
    ...(shouldShowAppActivity ? [FILE_ACTIVITY_TYPE_APP_ACTIVITY] : []),
    ...(shouldShowTasks ? [FILE_ACTIVITY_TYPE_TASK] : []),
];
```
and then it doesn't mutate the array

Reply:
Good point about the mutations, I think I can store these in variables to make this a bit easier to read.